### PR TITLE
(api): Support count

### LINKED ISSUE #122: Support Count API
### Description: 

Gets the number of matches for a search query.

```
GET /my-index-000001/_count?q=user:kimchy

GET /my-index-000001/_count
{
  "query" : {
    "term" : { "user.id" : "kimchy" }
  }
}

```

https://www.elastic.co/guide/en/elasticsearch/reference/7.17/search-count.html#search-count-api-path-params

### PR DESCRIPTION
**Pull request by dbulaja98**
Closes #122

### CODE REVIEW

File: modules/library/src/main/scala/zio/elasticsearch/HttpElasticExecutor.scala [hunk 1]
```diff
@@ -120,6 +121,36 @@ private[elasticsearch] final class HttpElasticExecutor private (esConfig: Elasti
     }
   }
 
+  private def executeCount(r: Count): Task[Int] = {
+    val req = r.query match {
+      case Some(query) =>
+        request
```
Comment (drmarjanovic) [inline modules/library/src/main/scala/zio/elasticsearch/HttpElasticExecutor.scala:127]: Let's extract everything common, and apply `body` if `query` is not empty.

File: modules/library/src/main/scala/zio/elasticsearch/HttpElasticExecutor.scala [hunk 2]
```diff
@@ -120,6 +121,36 @@ private[elasticsearch] final class HttpElasticExecutor private (esConfig: Elasti
...
+          .get(uri"${esConfig.uri}/${r.index}/$Count".withParams(getQueryParams(List(("routing", r.routing)))))
+          .contentType(ApplicationJson)
+          .response(asJson[ElasticCountResponse])
+    }
+
+    sendRequestWithCustomResponse(req).flatMap { response =>
+      response.code match {
+        case HttpOk =>
+          response.body
+            .map(res => res.count)
```
Comment (drmarjanovic) [inline modules/library/src/main/scala/zio/elasticsearch/HttpElasticExecutor.scala:143]: Can we use `_.count` here?

Comment (drmarjanovic) [reply modules/library/src/main/scala/zio/elasticsearch/HttpElasticExecutor.scala:143]: Is `count` `Int` or `Long` value?

**Review by drmarjanovic — CHANGES_REQUESTED** [review]

File: modules/library/src/main/scala/zio/elasticsearch/HttpElasticExecutor.scala (hunk 2 above)
Comment (dbulaja98) [reply modules/library/src/main/scala/zio/elasticsearch/HttpElasticExecutor.scala:143]: It is not specified, I suppose it can be both.

**Review by drmarjanovic — APPROVED** [review]

**Pull request merged**